src/basics.md

```diff
@@ -326,13 +327,185 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-run-piped-external-commands]: #ex-run-piped-external-commands
+<a name="ex-run-piped-external-commands"></a>
+## Run piped external commands
+
+[![clap-badge]][clap] [![cat-os-badge]][cat-os]
+
+Shows the `<N>`th biggest files/subdirectories in the specified
+`<directory>`. It spawns three Unix processes. It is equivalant to
+run: `du -ah <directory> | sort -hr | head -n <N>`.
+
+```rust,no_run
+# extern crate clap;
+
+# use clap::{App, Arg, ArgMatches};
+use std::error::Error;
+use std::fs::metadata;
+use std::process::{exit, Command, Stdio};
+
+static DU: &'static str = "du";
+static DU_ARGUMENTS: &'static str = "-ah";
+
+static SORT: &'static str = "sort";
+static SORT_ARGUMENTS: &'static str = "-hr";
+
+static HEAD: &'static str = "head";
+static HEAD_ARGUMENTS: &'static str = "-n";
+
+fn main() {
+    let arguments = declare_command_line_arguments();
+    let results = parse_number_of_results(&arguments);
+    let directory = parse_directory(&arguments);
+
+    let du_process = match Command::new(DU)
+        .arg(DU_ARGUMENTS)
+        .arg(directory)
+        .stdin(Stdio::null())
+        .stdout(Stdio::piped())
+        .spawn()
+    {
+        Err(why) => {
+            println!("couldn't spawn {}: {}", DU, why.description());
+            exit(1)
+        }
+        Ok(process) => process,
+    };
+
+    let du_stdout = du_process.stdout.unwrap_or_else(|| {
+        println!("Could not capture {} standard output.", DU);
+        exit(2);
+    });
+
+    let sort_process = match Command::new(SORT)
+        .arg(SORT_ARGUMENTS)
+        .stdin(du_stdout)
+        .stdout(Stdio::piped())
+        .spawn()
+    {
+        Err(why) => {
+            println!("couldn't spawn {}: {}", SORT, why.description());
+            exit(3)
+        }
+        Ok(process) => process,
+    };
+
+    let sort_stdout = sort_process.stdout.unwrap_or_else(|| {
+        println!("Could not capture {} standard output.", SORT);
+        exit(4);
+    });
+
+    let head_process = match Command::new(HEAD)
+        .arg(HEAD_ARGUMENTS)
+        .arg(results)
+        .stdin(sort_stdout)
+        .stdout(Stdio::piped())
+        .spawn()
+    {
+        Err(why) => {
+            println!("couldn't spawn {}: {}", HEAD, why.description());
+            exit(5)
+        }
+        Ok(process) => process,
+    };
+
+    let head_output = head_process.wait_with_output().unwrap_or_else(|err| {
+        println!("Could not capture {} standard output: {}", HEAD, err);
+        exit(6);
+    });
+
+
+    match String::from_utf8(head_output.stdout) {
```

Comment:
lets drop this whole boilerplate. and just perform the conversion from inside the `println!`

```rust
    println!(
        "Top {} biggest files and directories in {}\n{}",
        max_results,
        directory,
        String::from_utf8(head_output.stdout)?
    );
```